lib/std/process.zig

```diff
@@ -867,6 +867,15 @@ pub fn argsAlloc(allocator: Allocator) ![][:0]u8 {
     return result_slice_list;
 }
 
+test argsAlloc {
+    const allocator = std.testing.allocator;
+    const arg_iterator = try std.process.argsAlloc(allocator);
+    defer std.process.argsFree(allocator, arg_iterator);
+    for (arg_iterator) |arg| {
+        std.debug.print("{s}\n", .{arg});
```

Comment:
is stdout better or should tests not print at all?

Reply:
No printing at all.

Reply:
My intention with this PR was not to write a functional test for `argsAlloc`, but instead to document a basic example of how to use it, so that other beginners can learn more easily.

And since [`test`s show up as usage examples in the docs](https://ziglang.org/documentation/master/std/#A;std:json.parseFromSlice), I just tried to copy that approach. But I think this might be an inappropriate method of conveying an example, given the restrictions (which make sense for a unit test).

If there's a better way to document examples, please let me know. Otherwise, you can close this PR.

Reply:
Well, `std.Progress` just disables the test entirely but that doesn't seem good either:
https://github.com/ziglang/zig/blob/54d196bb300770394fd57db718ff1768edd552f8/lib/std/Progress.zig#L399-L405